app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantDetailBindingAdapter.kt

```diff
@@ -38,8 +38,8 @@ fun imageFromUrl(view: ImageView, imageUrl: String?) {
 }
 
 @BindingAdapter("goneIf")
-fun goneIf(view: View, isGone: Boolean) {
-    view.visibility = if (isGone) View.GONE else View.VISIBLE
+fun goneIf(view: FloatingActionButton, isGone: Boolean?) {
```

Comment:
I don't think ```goneIf``` should lose origin objective, because general ```views``` need it. 
On the contrary, I would like to add a new extension method for ```FloatingActionButton``` to hide.
How do you think? @andhie @tiembo

Reply:
Originally this `goneIf` did serve the purpose of a general `GONE` / `VISIBLE` binding adapter.  But since `goneIf` is currently only used with the FAB (and is in `PlantDetailBindingAdapter`) I think it's appropriate for the param to be a FAB instead of a `View`.  This also allows the use of `hide()` and `show()`, giving us animations for free.

If there's a need for a general purpose `goneIf` we can place it in a generic `BindingAdapter` file.

Reply:
at first i actually duplicated the exact same method name and bindingadapter. although Data Binding smartly knows which method to pick (`View` vs `FloatingActionButton` param), this causes the `View` goneif method to grey out as it was not used. plus, this is a sample app, i wish to keep it simple enough and less confusing to newbies